Task3-1.playground/Contents.swift

```diff
@@ -0,0 +1,157 @@
+/**
+ Create a new class Stack (see https://en.wikipedia.org/wiki/Stack_(abstract_data_type) for more details. It should have the following interface:
+
+ func push(element: Int) // Adds element on the top of the stack.
+ func pop() -> Int? // Removes and returns the top element of the stack. returns nil when the invoking stack is empty.
+ func top() -> Int? // Returns the top element of the stack, but does not remove it.
+
+ Create a new class StackStatistics that is a simple stack as described above but it also should support the following operation:
+
+ func minimumElement() -> Int? // Returns the first minimum element in the stack. Important: the function should work in O(1).
+
+ Use inheritance.
+ */
+
+/**
+ Update the hometask - use generics to allow your stack to work with different types.
+ Use opaque types if that's possible. Add a doc comments to describe interfaces and types.
+ */
+
+class Stack<Element> {
+    private var items = [Element]()
+
+    /// Adds the element on the top of the stack.
+    ///
+    /// Use this method to add a single element on the top of the stack.
+    ///
+    ///     var someStack = [1, 2, 3]
+    ///     someStack.push(element: 10)
+    ///     print(someStack)
+    ///     // Prints "[1, 2, 3, 10]"
+    ///
+    /// - Parameter element: The element to add on the top of the stack.
+    ///
+    /// - Complexity: O(1)
+    func push(element: Element) {
+        items.append(element)
+    }
+
+    /// Removes and returns the top element of the stack.
+    ///
+    /// Use this method to remove and return the top element of the stack.
+    /// Returns `nil` when the invoking stack is empty.
+    ///
+    /// - Returns: The top element of the stack if the stack is not
+    /// empty; otherwise, `nil`.
+    ///
+    /// - Complexity: O(1)
+    func pop() -> Element? {
+        if items.isEmpty {
+            return nil
+        } else {
+            return items.removeLast()
+        }
+    }
+
+    /// Returns the top element of the stack.
+    ///
+    /// If the stack is empty, the value of this property is `nil`.
```

Comment:
```suggestion
    /// If the stack is empty, the method returns `nil`.
```
may be slightly change the wording

Reply:
Thank you for your opinion.
I will think about it.

Reply:
Fixed.

Reply:
@dersim-davaod after the last change it looks much better